src/middlewared/middlewared/api/base/decorator.py

```diff
@@ -92,7 +93,7 @@ def wrapper(func):
         if skip_args is not None:
             func._skip_arg = skip_args
 
-        args_index = calculate_args_index(func, audit_callback)
+        args_index = calculate_args_index(func, pass_thread_local_storage, audit_callback)
```

Comment:
You can remove this `pass_thread_local_storage` check. I've got another PR that will fix it so that it's handled like the `_pass_app` attribute and we won't have to pass this argument to that function directly.

Reply:
I can't, it's not the last argument to check, it'll break `calculate_args_index`

Reply:
Unfortunately, we have conflicting PRs. Let me push my PR so you can understand where I'm coming from.